src/material-examples/autocomplete-display/autocomplete-display-example.ts

```diff
@@ -0,0 +1,43 @@
+import {Component} from '@angular/core';
+import {FormControl} from '@angular/forms';
+import {Observable} from 'rxjs/Observable';
+import 'rxjs/add/operator/startWith';
+import 'rxjs/add/operator/map';
+
+export class User {
+  constructor(public name: string) { }
+}
+
+@Component({
+  selector: 'autocomplete-display-example',
+  templateUrl: 'autocomplete-display-example.html',
+  styleUrls: ['autocomplete-display-example.css']
+})
+export class AutocompleteDisplayExample {
+
+  myControl = new FormControl();
+
+  options = [
+    new User('Mary'),
+    new User('Shelley'),
+    new User('Igor')
+  ];
+
+  filteredOptions: Observable<User[]>;
+
+  ngOnInit() {
+    this.filteredOptions = this.myControl.valueChanges
+        .startWith(null)
+        .map(user => user && typeof user === 'object' ? user.name : user)
+        .map(name => name ? this.filter(name) : this.options.slice());
+  }
+
+  filter(name: string): User[] {
+    return this.options.filter(option => new RegExp(`^${name}`, 'gi').test(option.name));
```

Comment:
Yeah, the issue here is that if someone types RegExp special characters into the input it will cause problems. Simpler to use `indexOf`.